src/Components/ShowProduct.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
+import LandingPage from "./LandingPage";
+import { useNavigate, useParams } from "react-router-dom";
+import { Toolbar } from "primereact/toolbar";
+import { DataTable, DataTableFilterMeta } from "primereact/datatable";
+import { Dropdown, DropdownChangeEvent } from "primereact/dropdown";
+import { Column } from "primereact/column";
+import { Panel } from "primereact/panel";
+import { Fieldset } from "primereact/fieldset";
+import { Button } from "primereact/button";
+import { FilterMatchMode } from "primereact/api";
+import { SplitButton } from "primereact/splitbutton";
+import TopContentShow from "./TopContentShow";
+
+interface Product {
+  jour?: string;
+  piece?: string;
+  reference?: string;
+  facture?: string;
+  type?: number;
+  compte_general?: string;
+  compte_tiers?: string;
+  libelle_ecriture?: string;
+  date_echeance?: string;
+  position_journal?: string;
+  debit?: string;
+  credit?: string;
+}
+interface VisibilityProps {
+  setProduct?: Dispatch<SetStateAction<Product>>;
+  product?: Product;
+  products?: Product[];
+  setProducts?: Dispatch<SetStateAction<Product[]>>;
+  setSelectedProduct?: Dispatch<SetStateAction<Product>>;
+  selectedProduct?: Product;
+}
+interface Type {
+  name?: string;
+  code: number | null;
+}
+
+const type: Type[] = [
+  { name: "Toutes les ecritures", code: null },
+  { name: "Ecritures lettrées", code: 1 },
+  { name: "Ecritures non lettrées", code: 2 },
+];
+const produit: Product[] = [
```

Comment:
why you don't make that in file call like database/fake_data.tsx

Reply:
I hadn't thought about it, I would do it